### PR TITLE
Feature/tdd

### PR DESCRIPTION
**Pull request by isaka-lumato**
## TDD with RSpec

- used tdd approach
- following good testing practices

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+### Prerequisites
+```
+Ruby
+```
+
+### Setup
+```
+git clone git@github.com:isaka-lumato/tdd_ruby.git
+cd tdd_ruby.git
+```
```
Comment (Dipeshtwis) [inline README.md:29]: - [ ] It's a small change but it is very important for you to provide all the commands about your project for the other users. In the README file, Please mention the step on how to test the project. [HInt: `rspec`]

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require_relative '../solver'
+
+describe Solver do
+  describe 'factorial' do
+    it 'if is 2 it shoulkd return 2' do
+      expect(subject.factorial(2)).to eq 2
+    end
+    it 'should be 1 uf we pass 0' do
+      expect(subject.factorial(0)).to eq(1)
```
Comment (Dipeshtwis) [inline spec/solver_spec.rb]: - [ ] There is some typo in your sentence, please fix that. 

 - Instead of `shoulkd` ~> `should`
 - Instead of `uf` ~> `if`

Comment (Dipeshtwis) [inline spec/solver_spec.rb]: [OPTIONAL]

- I think it is better to test factorial for greater numbers as well like 5 or 8.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi Team, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: All tests are passing. :heavy_check_mark: 
:+1: Handles special case as well.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by oloomoses — APPROVED** [review]
Hi team, 

Well done on making all the suggested changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**